src/lib/delete/deleteItem.ts

```diff
@@ -0,0 +1,168 @@
+import { HttpStatusCode, IModify, IRead } from '@rocket.chat/apps-engine/definition/accessors';
+import { IRoom } from '@rocket.chat/apps-engine/definition/rooms';
+import {
+  UIKitBlockInteractionContext,
+  UIKitViewSubmitInteractionContext,
+} from '@rocket.chat/apps-engine/definition/uikit';
+
+import { TodoistApp } from '../../../TodoistApp';
+import { MiscEnum } from '../../enums/Misc';
+import { getCommentUrl, getLabelUrl, getSectionUrl, getTaskUrl } from '../../helpers/const';
+import { deleteConfirmationModal } from '../../modals/deleteConfirmationModal';
+
+interface DeleteItemParams {
+  app: TodoistApp;
+  context: UIKitViewSubmitInteractionContext;
+  room: IRoom;
+  read: IRead;
+  modify: IModify;
+  getUrl: (id?: string) => string;
+  itemType: string;
+}
+
+interface DeleteHandlerParams {
+  context: UIKitViewSubmitInteractionContext;
+  room: IRoom;
+  read: IRead;
+  modify: IModify;
+  app?: TodoistApp;
+}
+
+export async function deleteItem({
+  app,
+  context,
+  room,
+  read,
+  modify,
+  getUrl,
+  itemType,
+}: DeleteItemParams) {
+  const data = context.getInteractionData();
+  const itemId = data.view.submit!.value;
+  const user = data.user;
+  const roomId = room.id;
+  let roomFromId = await read.getRoomReader().getById(roomId);
+  if (!room) room = roomFromId!;
+  try {
+    const response = await app.getHttpHelperInstance().delete(user, getUrl(itemId));
```

Comment:
please wrap this in a service call so you don't have to handle with http codes inside a domain context